src/Microsoft.DotNet.Build.CloudTestTasks/SimulateHelixExecution.cs

```diff
@@ -0,0 +1,420 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Newtonsoft.Json;
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    /// <summary>
+    /// Simulates Helix Job execution using real Helix Client scripts, on the local machine.
+    /// Currently this includes downloading the URLs listed in the helix job list, since it's a fair amount of extra
+    /// work to make a job list with local file paths, plus this can be a source of problems to debug.
+    /// </summary>
+    public sealed class SimulateHelixExecution : Task
+    {
+        /// <summary>
+        /// Job-list JSON file describing Helix Work Item(s)
+        /// </summary>
+        [Required]
+        public ITaskItem JobList { get; set; }
+
+        /// <summary>
+        /// For Job lists with greater than one work item, an optional string to filter execution down to a single test.
+        /// </summary>
+        public string WorkItemExecutionFilter { get; set; }
+
+        /// <summary>
+        /// Base folder path for executing jobs.  Will be created if it does not exist.
+        /// </summary>
+        public string ExecutionFolder { get; set; }
+
+        /// <summary>
+        /// Python executable for runner scripts. Will attempt to find it in default places if not provided, but cannot continue without it.
+        /// </summary>
+        public string PythonExecutable { get; set; }
+
+        /// <summary>
+        /// Optional CorrelationId value for run.  Set to a constant value to prevent repeated unzipping, but may leave stale files. 
+        /// </summary>
+        public string CorrelationId { get; set; }
+
+        /// <summary>
+        /// Input items with metadata stripped out.
+        /// </summary>
+        [Output]
+        public ITaskItem[] ExecutionLogs { get; set; }
+
+        // Private constants
+        private const string DefaultPythonPathWindows = @"c:\Python27\python.exe";
+        private const string DefaultPythonPathUnix = "/usr/bin/python";
```

Comment:
Obviously I'm not a fan of the defaults here but I see that you also have the properties set to pass them in. Should this defaults be properties as well? I ask because given the case we need to change the defaults for whatever reason, it will take a buildtools update, as opposed to a targets file change in corefx side. This applies to the HelixUris as well.